### DISCUSSION TITLE
plugin idea - agentless SSH workers

### DISCUSSION THREAD

**Discussion by mikeTWC1984** [jhuckaby/Cronicle#445] [Ideas]
I think  that somewhat easy to implement, but could be very powerful. It could look exactly the same as regular script plugin, just with an option to select server (to connect over ssh). The nice bonus - it would work just fine with N workers behind loadbalancer, good scenario for swarm/kubernetes - you can just define 1 worker and then create as many replicas as needed.

**Comment by mcfetz** [comment]
I already use the ssh remote command feature in the shell plugin to run scripts on a server who doesn't run a cronicle worker.

Plugin: Shell
Command: ssh -o StrictHostKeyChecking=no [remote user]@[remote host] "[remote command]"

**Reply by mikeTWC1984** [reply]
Yeah, you definitely can do this. What I have in mind is the same thing as script plugin - you supply a script with shebang, then cronicle  takes care of  auth, copying it to target machine and executing. Or even run this on a server group like ansible.
